Libraries/LLM/Llama.swift

```diff
@@ -12,9 +12,9 @@ private class Attention: Module {
     let args: LlamaConfiguration
     let scale: Float
 
-    @ModuleInfo(key: "q_proj") var wq: Linear
+    @ModuleInfo(key: "q_proj") var wq: LoRAReplacableLinear
```

Comment:
We have to use a different type because we are going to replace the `Linear` with a `LoRALinear` which is not a sub-type of `Linear`.

- [ ] document this

Reply:
Yes, it is possible to do that -- this was mimicking the structure from the python code.  Now that I see how all the serialization works, it would be doable.

Reply:
Slightly trickier than anticipated -- I was thinking I could just do something like QuantizedLinear where it is Linear + scales + biases.  That would work fine, but I need the same thing for QuantizedLinear too.  It works much better as a LoRALinear which has-a Linear (or any subtype such as QuantizedLinear).

The other option would be to have LoRALinear and LoRAQuantizedLinear.  That isn't too bad since most of the code is in the toLinear() method and that has to deal with QuantizedLinear anyway.

Reply:
I didn't quite follow the first option. Second option seems ok. Maybe another option is something like a LoRA Linear that takes a union [QuantizedLinear, Linear]?

Reply:
> Maybe another option is something like a LoRA Linear that takes a union [QuantizedLinear, Linear]?

That is what we have now, but the sub typing doesn't work right if it is also is-a Linear (instead of LoRAReplacableLinear).  LoRAReplacableLinear is roughly union [Linear, QuantizedLinear, LoraLinear] as expressed in swift.

If we were to have a single type, LoRALinear that is-a Linear we would have to have both has-a Linear and also is-a Linear (two sets of weights).  I think that would be a mess.

If that didn't make sense, consider how the call will work:

```swift
    public override func callAsFunction(_ x: MLXArray) -> MLXArray {
        let y = super.callAsFunction(x.asType(weight.dtype))
        let z = matmul(matmul(x, self.loraA), self.loraB)
        return y + scale * z
    }
```

If super can't be both a `Linear` and a `QuantizedLinear`.

Anyway, it probably doesn't matter -- the first option looks like it will work fine.  I am just getting it all hooked up right now.

Reply:
This is complete -- it is mostly the same from the caller's point of view but now the layers are subclasses of Linear and you don't need anything special in the host model.